python/ray/serve/_private/replica.py

```diff
@@ -977,17 +936,34 @@ async def call_user_method_generator(
             assert (
                 not request_metadata.is_http_request
             ), "HTTP requests should go through `call_user_method`."
+
             user_method = self.get_runner_method(request_metadata)
+            if request_metadata.is_grpc_request:
+                assert len(request_args) == 1 and isinstance(
+                    request_args[0], gRPCRequest
+                )
+                request_args = pickle.loads(request_args[0].grpc_user_request)
```

Comment:
This might need to be a tuple :)

Reply:
sigh, sometimes I really wish we had a compiler :)